README.md
# NAME

Badge::Depot::Plugin::Gratipay - Gratipay plugin for Badge::Depot (deprecated)

<div>
    <p>
    <img src="https://img.shields.io/badge/perl-5.10.1+-blue.svg" alt="Requires Perl 5.10.1+" />
    <a href="https://travis-ci.org/Csson/p5-Badge-Depot-Plugin-Gratipay"><img src="https://api.travis-ci.org/Csson/p5-Badge-Depot-Plugin-Gratipay.svg?branch=master" alt="Travis status" /></a>
    <a href="http://cpants.cpanauthors.org/release/CSSON/Badge-Depot-Plugin-Gratipay-0.0103"><img src="http://badgedepot.code301.com/badge/kwalitee/CSSON/Badge-Depot-Plugin-Gratipay/0.0103" alt="Distribution kwalitee" /></a>
    <a href="http://matrix.cpantesters.org/?dist=Badge-Depot-Plugin-Gratipay%200.0103"><img src="http://badgedepot.code301.com/badge/cpantesters/Badge-Depot-Plugin-Gratipay/0.0103" alt="CPAN Testers result" /></a>
    <img src="https://img.shields.io/badge/coverage-96.7%-yellow.svg" alt="coverage 96.7%" />
    </p>
</div>

# VERSION

Version 0.0103, released 2018-01-29.

# SYNOPSIS

    use Badge::Depot::Plugin::Gratipay;

    my $badge = Badge::Depot::Plugin::Gratipay->new(user => 'my_name');

    print $badge->to_html;
    # prints '<a href="https://gratipay.com/my_name"><img src="https://img.shields.io/my_name.svg" /></a>'

# STATUS

Deprecated. Since [Gratipay](https://gratipay.com) has shut down this distribution has no purpose.

# DESCRIPTION

Create a [Gratipay](https://gratipay.com) badge for a Gratipay user.

This class consumes the [Badge::Depot](https://metacpan.org/pod/Badge::Depot) role.

# ATTRIBUTES

## user

The Gratipay user name.

## custom\_image\_url

By default, this module shows an image from [shields.io](https://shields.io). Use this attribute to override that with a custom url. Use a `%s` placeholder where the user name should be inserted.

# SEE ALSO

- [Badge::Depot](https://metacpan.org/pod/Badge::Depot)
- [Task::Badge::Depot](https://metacpan.org/pod/Task::Badge::Depot)

# SOURCE

[https://github.com/Csson/p5-Badge-Depot-Plugin-Gratipay](https://github.com/Csson/p5-Badge-Depot-Plugin-Gratipay)

# HOMEPAGE

[https://metacpan.org/release/Badge-Depot-Plugin-Gratipay](https://metacpan.org/release/Badge-Depot-Plugin-Gratipay)

# AUTHOR

Erik Carlsson <info@code301.com>

# COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Erik Carlsson.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for Badge-Depot-Plugin-Gratipay

{{$NEXT}}
 [API Changes]

 [Bug Fixes]

 [Documentation]

 [Enhancements]

 [New Features]

 [Other Changes]

 [Requirements]

 [Stats & Dependency Changes]

 [Testing]

0.0103 2018-01-29T19:03:57Z UTC
 - Deprecated, due to Gratipay having shut down.

 [Stats & Dependency Changes]
 - 3 files changed, 9 insertions(+), 9 deletions(-)
 - (run req) ~ Types::Standard (any) --> 1.000006
 - (run req) ~ namespace::autoclean (any) --> 0.22
 - (test req) + IO::Handle (any)
 - (test req) + IPC::Open3 (any)
 - (conf req) + ExtUtils::MakeMaker (any)
 - (conf req) - Module::Build::Tiny
 - (dev req) + Badge::Depot 0.0103
 - (dev req) + Badge::Depot::Plugin::Coverage (any)
 - (dev req) + Badge::Depot::Plugin::Cpantesters (any)
 - (dev req) + Badge::Depot::Plugin::Kwalitee (any)
 - (dev req) + Badge::Depot::Plugin::Perl (any)
 - (dev req) + Badge::Depot::Plugin::Travis (any)
 - (dev req) + Dist::Zilla 6.010
 - (dev req) + Dist::Zilla::Plugin::Authority 1.009
 - (dev req) + Dist::Zilla::Plugin::BumpVersionAfterRelease::Transitional
   0.008
 - (dev req) + Dist::Zilla::Plugin::ChangeStats::Dependencies::Git 0.0200
 - (dev req) + Dist::Zilla::Plugin::Clean 0.07
 - (dev req) + Dist::Zilla::Plugin::Deprecated (any)
 - (dev req) + Dist::Zilla::Plugin::DistIller::MetaGeneratedBy (any)
 - (dev req) + Dist::Zilla::Plugin::Git 2.041
 - (dev req) + Dist::Zilla::Plugin::Git::CheckFor::CorrectBranch 0.014
 - (dev req) + Dist::Zilla::Plugin::Git::Commit (any)
 - (dev req) + Dist::Zilla::Plugin::Git::Contributors 0.030
 - (dev req) + Dist::Zilla::Plugin::InstallRelease 0.008
 - (dev req) + Dist::Zilla::Plugin::MakeMaker (any)
 - (dev req) + Dist::Zilla::Plugin::MetaConfig (any)
 - (dev req) + Dist::Zilla::Plugin::NextRelease::Grouped 0.0200
 - (dev req) + Dist::Zilla::Plugin::PromptIfStale 0.054
 - (dev req) + Dist::Zilla::Plugin::RewriteVersion::Transitional 0.007
 - (dev req) + Dist::Zilla::Plugin::Test::Compile 2.057
 - (dev req) + Dist::Zilla::Plugin::Test::EOF 0.0501
 - (dev req) + Dist::Zilla::Plugin::Test::EOL 0.18
 - (dev req) + Dist::Zilla::Plugin::Test::NoTabs 0.15
 - (dev req) + Dist::Zilla::Plugin::Test::Version 1.05
 - (dev req) + Pod::Weaver::Section::Badges 0.0401
 - (dev req) + Pod::Weaver::Section::GenerateSection 1.01
 - (dev req) + Test::EOF (any)
 - (dev req) + Test::EOL (any)
 - (dev req) + Test::More 0.88
 - (dev req) + Test::NoTabs 1.4
 - (dev req) + Test::Version 1
 - (dev req) ~ Dist::Zilla::Plugin::CheckChangesHasContent (any) --> 0.008
 - (dev req) ~ Dist::Zilla::Plugin::CopyFilesFromBuild (any) --> 0.170880
 - (dev req) ~ Dist::Zilla::Plugin::GithubMeta (any) --> 0.54
 - (dev req) ~ Dist::Zilla::Plugin::MetaProvides::Package (any) -->
   2.004003
 - (dev req) ~ Dist::Zilla::Plugin::PodWeaver (any) --> 4.006
 - (dev req) ~ Dist::Zilla::Plugin::ReadmeAnyFromPod (any) --> 0.163250
 - (dev req) ~ Dist::Zilla::Plugin::RunExtraTests (any) --> 0.028
 - (dev req) ~ Dist::Zilla::Plugin::Test::ReportPrereqs (any) --> 0.027
 - (dev req) ~ Dist::Zilla::Plugin::TravisYML (any) --> 1.15
 - (dev req) ~ Test::Warnings (any) --> 0.026
 - (dev req) - Dist::Zilla::Plugin::CheckExtraTests
 - (dev req) - Dist::Zilla::Plugin::ContributorsFromGit
 - (dev req) - Dist::Zilla::Plugin::MetaProvides::Class
 - (dev req) - Dist::Zilla::Plugin::ModuleBuildTiny
 - (dev req) - Dist::Zilla::Plugin::NextRelease
 - (dev req) - Dist::Zilla::Plugin::NextVersion::Semantic
 - (dev req) - Dist::Zilla::Plugin::OurPkgVersion
 - (dev req) - Dist::Zilla::Plugin::PodnameFromClassname
 - (dev req) - Dist::Zilla::Plugin::Prereqs::Plugins
 - (dev req) - Dist::Zilla::Plugin::PreviousVersion::Changelog
 - (dev req) - Dist::Zilla::Plugin::ReversionOnRelease
 - (dev req) - Pod::Weaver::Section::Homepage::DefaultCPAN
 - (dev req) - Pod::Weaver::Section::Source::DefaultGitHub
 - (dev sug) + Dist::Iller 0.1408
 - (dev sug) + Dist::Iller::Config::Author::CSSON 0.0319

0.0102 2015-03-19T16:06:14Z CET
 [Documentation]
 - Added Task::Badge::Depot under see also.

 [Requirements]
 - Listed Types::URI as a prereq.

0.0101 2015-03-05T23:17:57Z CET
 [Documentation]
 - Added link to github repo in pod.

0.0100 2015-02-28T18:50:55Z CET
 - Initial release


lib/Badge/Depot/Plugin/Gratipay.pm
use strict;
use warnings;

package Badge::Depot::Plugin::Gratipay;

use Moose;
use namespace::autoclean;
use Types::Standard qw/Str/;
use Types::URI qw/Uri/;
with 'Badge::Depot';

our $VERSION = '0.0104';
# ABSTRACT: Gratipay plugin for Badge::Depot (deprecated)
# AUTHORITY

has user => (
    is => 'ro',
    isa => Str,
    required => 1,
);
has custom_image_url => (
    is => 'ro',
    isa => Uri,
    coerce => 1,
    default => 'https://img.shields.io/gratipay/%s.svg',
);


sub BUILD {
    my $self = shift;
    $self->link_url(sprintf 'https://gratipay.com/%s', $self->user);
    $self->image_url(sprintf $self->custom_image_url, $self->user);
    $self->image_alt('Gratipay');
}

1;

__END__


=pod

=head1 SYNOPSIS

    use Badge::Depot::Plugin::Gratipay;

    my $badge = Badge::Depot::Plugin::Gratipay->new(user => 'my_name');

    print $badge->to_html;
    # prints '<a href="https://gratipay.com/my_name"><img src="https://img.shields.io/my_name.svg" /></a>'

=head1 STATUS

Deprecated. Since L<Gratipay|https://gratipay.com> has shut down this distribution has no purpose.

=head1 DESCRIPTION

Create a L<Gratipay|https://gratipay.com> badge for a Gratipay user.

This class consumes the L<Badge::Depot> role.

=head1 ATTRIBUTES

=head2 user

The Gratipay user name.

=head2 custom_image_url

By default, this module shows an image from L<shields.io|https://shields.io>. Use this attribute to override that with a custom url. Use a C<%s> placeholder where the user name should be inserted.

=head1 SEE ALSO

=for :list
* L<Badge::Depot>
* L<Task::Badge::Depot>

=cut


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.010.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Gratipay plugin for Badge::Depot (deprecated)",
  "AUTHOR" => "Erik Carlsson <info\@code301.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Badge-Depot-Plugin-Gratipay",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "Badge::Depot::Plugin::Gratipay",
  "PREREQ_PM" => {
    "Badge::Depot" => 0,
    "Moose" => "2.0000",
    "Types::Standard" => "1.000006",
    "Types::URI" => 0,
    "namespace::autoclean" => "0.22"
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.96"
  },
  "VERSION" => "0.0104",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Badge::Depot" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => "2.0000",
  "Test::More" => "0.96",
  "Types::Standard" => "1.000006",
  "Types::URI" => 0,
  "namespace::autoclean" => "0.22"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
